cmd/kubeadm/app/phases/upgrade/staticpods_test.go

```diff
@@ -208,33 +209,50 @@ func (spm *fakeStaticPodPathManager) BackupEtcdDir() string {
 	return spm.backupEtcdDir
 }
 
-type fakeTLSEtcdCluster struct{ TLS bool }
+type fakeTLSEtcdClient struct{ TLS bool }
 
-func (cluster fakeTLSEtcdCluster) HasTLS() (bool, error) {
-	return cluster.TLS, nil
+func (c fakeTLSEtcdClient) HasTLS() bool {
+	return c.TLS
 }
 
-func (cluster fakeTLSEtcdCluster) GetStatus() (*clientv3.StatusResponse, error) {
+func (c fakeTLSEtcdClient) GetStatus() (*clientv3.StatusResponse, error) {
 	client := &clientv3.StatusResponse{}
 	client.Version = "3.1.12"
 	return client, nil
 }
 
-type fakePodManifestEtcdCluster struct{ ManifestDir, CertificatesDir string }
+func (c fakeTLSEtcdClient) WaitForStatus(delay time.Duration, retries int, retryInterval time.Duration) (*clientv3.StatusResponse, error) {
+	return c.GetStatus()
+}
+
+type fakePodManifestEtcdClient struct{ ManifestDir, CertificatesDir string }
 
-func (cluster fakePodManifestEtcdCluster) HasTLS() (bool, error) {
-	return etcdutil.PodManifestsHaveTLS(cluster.ManifestDir)
+func (c fakePodManifestEtcdClient) HasTLS() bool {
+	hasTLS, _ := etcdutil.PodManifestsHaveTLS(c.ManifestDir)
+	return hasTLS
```

Comment:
for the test, we should add a TLS var to the struct and implement `NewFakePodManifestEtcdClient()` that pre-initializes it so that it behaves more like `etcdutil.GenericClient` (which does not handle podManifest mutations without recreation of the client)

Reply:
Nearing EOD here, but will try to see if I can find some time tonight to try and improve the tests around this.